src/lib/core/WeaveFabricState.cpp

```diff
@@ -113,6 +113,10 @@ void WeaveSessionKey::Init(void)
     NodeId = kNodeIdNotSpecified;
     NextMsgId.Init(0);
     MaxRcvdMsgId = 0;
+    InitialSendMsgId = 0;
```

Comment:
I think these Initial* fields are redundant and Resumption message Ids should be calculated using NextMsgId value.

Reply:
I initially did that. But Cloud uses the initial values to calculate the resumption msgIds. The reasons range from having to do less lookups/writes to keep track of last messageId to having a resumption messageId available when the device connects a second time while the Cloud side is still holding on to its old connection.